File: src/AutoContext.Worker.Web/src/types/log-entry.ts
/**
 * Log severity levels. Names match the .NET `LogLevel` enum so the
 * extension-side LogServer can route both .NET and Node worker output
 * through the same `levelToMethod` mapper.
 */
export type LogLevel = 'Trace' | 'Debug' | 'Information' | 'Warning' | 'Error' | 'Critical';

/**
 * In-process record enqueued by every {@link PipeLogger} call. Drained
 * off-thread by `LoggingClient` and either sent over the pipe (the
 * extension is listening) or written to stderr (fallback).
 *
 * TypeScript counterpart of the C# `LogEntry` readonly struct in
 * `AutoContext.Worker.Logging`.
 */
export interface LogEntry {
    readonly category: string;
    readonly level: LogLevel;
    readonly message: string;
    readonly exception?: string;
    readonly correlationId?: string;
}

/**
 * Wire shape for one NDJSON log record streamed from a worker to the
 * extension's LogServer. Property names are intentionally lowercased
 * to keep the serialized payload compact and to match the
 * `AutoContext.Worker.Shared` C# `JsonLogEntry`.
 */
export interface JsonLogEntry {
    readonly category: string;
    readonly level: string;
    readonly message: string;
    readonly exception?: string;
    readonly correlationId?: string;
}

/**
 * Wire shape for the greeting line every `LoggingClient` sends as
 * the very first NDJSON line on the pipe — lets the extension route
 * subsequent records to the per-worker output channel.
 */
export interface JsonLogGreeting {
    readonly clientName: string;
}

File: src/AutoContext.Framework.Logging/LogEntry.cs
namespace AutoContext.Framework.Logging;

using Microsoft.Extensions.Logging;

/// <summary>
/// In-process record enqueued by every <see cref="ILogger"/> call routed
/// through <see cref="PipeLogger"/>. Drained off-thread by
/// <see cref="LoggingClient"/> and either sent over the pipe (extension
/// is listening) or written to stderr (fallback).
/// </summary>
/// <remarks>
/// <c>EventId</c> is intentionally omitted from the v1 wire format — the
/// extension-side LogServer presents records as plain text in a
/// <c>LogOutputChannel</c>, where event ids carry no UI affordance.
/// Extending <see cref="LogEntry"/> and <c>JsonLogEntry</c> to carry an
/// event id (and matching the change in the LogServer reader) is the
/// obvious next step if structured filtering arrives later.
/// </remarks>
public readonly record struct LogEntry(
    string Category,
    LogLevel Level,
    string Message,
    Exception? Exception,
    string? CorrelationId);
